src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableOrchestrationContext.cs

```diff
@@ -1002,9 +1016,22 @@ internal void SendEntityMessage(OrchestrationInstance target, string eventName,
                     eventContent);
             }
 
+            this.IncrementActionsOrThrowException();
             this.InnerContext.SendEvent(target, eventName, eventContent);
         }
 
+        private void IncrementActionsOrThrowException()
+        {
+            if (actionCount >= this.maxActionCount)
+            {
+                throw new InvalidOperationException("Maximum amount of orchestration actions (" + this.maxActionCount + ") has been reached. This value can be configured in local.settings.json file as MaxOrchestrationActions.");
```

Comment:
`local.settings.json` only applies to local development. It does not apply to azure deployments. But shouldn't this actually be configured in host.json?